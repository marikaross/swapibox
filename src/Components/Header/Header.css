.header {
  background-color: black;
  display: block; }

img {
  width: 30%;
  margin: 30px;
  display: block;
  margin: 0 auto; }

.longAgo {
  color: #56C0D4; }

.planet-button,
.people-button,
.vehicles-button,
.favorites {
  background-color: black;
  border: 2px solid #f8e654;
  border-radius: 12px;
  color: #f8e654;
  font-size: 1.5em;
  margin: 10px; }
