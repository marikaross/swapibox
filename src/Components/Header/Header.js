import React from 'react';
import PropTypes from 'prop-types';
import './Header.css';

const Header = (props) => {
  return (
    <div className="header">
      <h2></h2>
      <button className="people-button" onClick={props.getPeople}>People</button>
      <button className="planet-button" onClick={props.getPlanets}>Planets</button>
      <button className="vehicles-button" onClick={props.getStarships}>Starships</button>
      <button className='favorites'>Favorites</button>
    </div>
    )
}

export default Header;

Header.propTypes = {
  getPeople: PropTypes.func.isRequired,
  getPlanets: PropTypes.func.isRequired,
  getStarships: PropTypes.func.isRequired
}