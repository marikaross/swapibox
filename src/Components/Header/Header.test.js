import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header.js';

describe('header', () => {
  it('should call getPeople onClick', () => {
    const mockGetPeople = jest.fn();
    const wrapper = shallow(<Header getPeople={mockGetPeople}/>)

    wrapper.find('.people-button').simulate('click');
    expect(mockGetPeople).toBeCalled();
  });

  it('should call getPlanets onClick', () => {
    const mockGetPlanets = jest.fn();
    const wrapper = shallow(<Header getPlanets={mockGetPlanets} />)

    wrapper.find('.planet-button').simulate('click');
    expect(mockGetPlanets).toBeCalled();
  });

  it('should call getStarships onClick', () => {
    const mockGetStarships = jest.fn()
    const wrapper = shallow(<Header getStarships={mockGetStarships} />)
    wrapper.find('.vehicles-button').simulate('click');
  })


  it('should match the snapshot with all data passed in correctly', () => {

    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();

  })
})