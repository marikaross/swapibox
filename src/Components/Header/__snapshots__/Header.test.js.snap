// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <button
          className="people-button"
          onClick={undefined}
>
          People
</button>,
        <button
          className="planet-button"
          onClick={undefined}
>
          Planets
</button>,
        <button
          className="vehicles-button"
          onClick={undefined}
>
          Starships
</button>,
        <button
          className="favorites"
>
          Favorites
</button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "people-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "planet-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Starships",
          "className": "vehicles-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Starships",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
          "className": "favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <button
            className="people-button"
            onClick={undefined}
>
            People
</button>,
          <button
            className="planet-button"
            onClick={undefined}
>
            Planets
</button>,
          <button
            className="vehicles-button"
            onClick={undefined}
>
            Starships
</button>,
          <button
            className="favorites"
>
            Favorites
</button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "planet-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Starships",
            "className": "vehicles-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Starships",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
